Add omdb command to forcefully quiesce db_metadata_nexus records

smklein opened the pull request:
Builds on https://github.com/oxidecomputer/omicron/pull/9034

Adds a command to handle permanent Nexus failure during handoff.

Fixes #9008

davepacheco commented on dev-tools/omdb/src/bin/omdb/db/db_metadata.rs:
```diff
@@ -27,8 +31,37 @@ pub struct DbMetadataArgs {
+    /// This operation is intended to assist in the explicit case where a Nexus
+    /// is unable to finish marking itself quiesced during the handoff process,
+    /// and cannot be expunged.
+    ForceNexusQuiesce(ForceNexusQuiesceArgs),
```
Take it or leave it:

```suggestion
    ForceMarkNexusQuiesced(ForceMarkNexusQuiescedArgs),
```

The idea is to convey that you're _marking_ Nexus quiesced, whether it is or not -- and _not_ that you're actually quiescing anything.

smklein replied:
Updated!

davepacheco reviewed (approved):
It would be nice to do a confirmation and maybe an attempt to check whether Nexus is up, similar to what we do for dangerous saga operations:

https://github.com/oxidecomputer/omicron/blob/3d2f1182cbb0426bbafdc27a6bfa31bc585aeba2/dev-tools/omdb/src/bin/omdb/db/saga.rs#L173-L226

There are enough warning signs here, and I don't think this is likely a tool that people will stumble on accidentally, that I wouldn't call this a blocker.

smklein commented on dev-tools/omdb/src/bin/omdb/db/db_metadata.rs:
```diff
@@ -27,8 +31,37 @@ pub struct DbMetadataArgs {
+    /// Forcing Nexus to quiesce is already an unsafe operation; this makes
+    /// it even less safe. Use with caution.
+    #[arg(long, action=ArgAction::SetTrue)]
+    ignore_target_blueprint: bool,
```
I renamed this to `skip_blueprint_validation`, but have updated the comment

smklein commented:
> It would be nice to do a confirmation and maybe an attempt to check whether Nexus is up, similar to what we do for dangerous saga operations:
> 
> https://github.com/oxidecomputer/omicron/blob/3d2f1182cbb0426bbafdc27a6bfa31bc585aeba2/dev-tools/omdb/src/bin/omdb/db/saga.rs#L173-L226
> 
> There are enough warning signs here, and I don't think this is likely a tool that people will stumble on accidentally, that I wouldn't call this a blocker.

I went ahead and incorporated a confirmation prompt, but this breaks automated tests, so I also added an optional `skip_confirmation` argument.
